### PR TITLE
feat: search ML features frontend components

### PR DESCRIPTION
**Pull request by allisonsuarez**
### Summary of Changes

Implemented all of the UI functionality to include ML Features as a resource in search, API work will be in a separate PR.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/amundsen_application/static/js/interfaces/Resources.ts [hunk 1]
```diff
@@ -55,6 +45,21 @@ export interface TableResource extends Resource {
...
+  version: string;
+  feature_group: string;
+  entity: string;
+  data_type?: string;
+  description: string;
+  availability: string[];
+  last_updated_timestamp?: number;
+  created_timestamp?: number;
+  badges?: Badge[];
+}
```
Comment (danwom) [inline frontend/amundsen_application/static/js/interfaces/Resources.ts:61]: The FeatureResource is the data type returned by search results. Are all of these fields part of the search result?

File: frontend/amundsen_application/static/js/interfaces/Feature.ts [hunk 2]
```diff
@@ -43,3 +39,9 @@ export interface FeatureSummary {
 export interface FeatureStats {
   name: string;
 }
+
+export interface FeatureWatermark {
+  key: string;
+  watermark_type: string;
+  time: string;
+}
```
Comment (danwom) [inline frontend/amundsen_application/static/js/interfaces/Feature.ts:47]: Is the different watermark format inherited from some upstream source or are have we created this format ourselves? I'd prefer to reuse the existing watermark, or to change both of them to the new format. WDYT?

File: frontend/amundsen_application/static/js/config/config-default.ts [hunk 3]
```diff
@@ -241,6 +203,42 @@ const configDefault: AppConfig = {
...
+        hive: {
+          displayName: 'Hive',
+          iconClass: 'icon-hive',
+        },
+      },
+      filterCategories: [
+        {
+          categoryId: 'entity',
+          displayName: 'Entity',
+          helpText: 'Enter exact tag name or a regex wildcard pattern',
```
Comment (danwom) [inline frontend/amundsen_application/static/js/config/config-default.ts:218]: `helpText` for this and others should be updated. `tag` reference should only be on the tag filter.

File: frontend/amundsen_application/static/js/interfaces/Resources.ts (hunk 1 above)
Comment (allisonsuarez) [reply frontend/amundsen_application/static/js/interfaces/Resources.ts:61]: I am still waiting on Dmitriy to get search done so unclear so far

File: frontend/amundsen_application/static/js/interfaces/Feature.ts (hunk 2 above)
Comment (allisonsuarez) [reply frontend/amundsen_application/static/js/interfaces/Feature.ts:47]: We could map the new feature watermark model dmitriy made to the existing one in FE I believe, but there is no column name

**Review by Golodhros — APPROVED** [review]

**Comment by allisonsuarez** [comment]
@devtagare

**Pull request merged**